### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by max-prtsr** [eslint/eslint#16973]
### What rule do you want to change?

curly

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// allow return
if (condition) return "something"

// allow throw
if (condition) throw new Error("Something bad")
```

### What does the rule currently do for this code?

`curly: error` marks them as invaid

### What will the rule do after it's changed?

`curly: error` ignores them

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Maybe use a regexp to ignore the operator that must be block?
e.g.
```js
curly: ["error", "all", {ignorePattern: "^(return|throw)"]
```

**Comment by mdjermanovic** [comment]
Hi @max-prtsr, thanks for the issue!

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we are not adding new options to the `curly` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by max-prtsr** [comment]
@mdjermanovic sad, but thank you for a reply
